awsutil_v2/awsutil_v2.go

```diff
@@ -0,0 +1,187 @@
+package awsutil_v2
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/credentials/stscreds"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/mitchellh/mapstructure"
+)
+
+// Some AWS destinations are using SecretAccessKey instead of accessKey
+type SessionConfig struct {
+	Region              string         `mapstructure:"region"`
+	AccessKeyID         string         `mapstructure:"accessKeyID"`
+	AccessKey           string         `mapstructure:"accessKey"`
+	SecretAccessKey     string         `mapstructure:"secretAccessKey"`
+	SessionToken        string         `mapstructure:"sessionToken"`
+	RoleBasedAuth       bool           `mapstructure:"roleBasedAuth"`
+	IAMRoleARN          string         `mapstructure:"iamRoleARN"`
+	ExternalID          string         `mapstructure:"externalID"`
+	WorkspaceID         string         `mapstructure:"workspaceID"`
+	Endpoint            *string        `mapstructure:"endpoint"`
+	S3ForcePathStyle    *bool          `mapstructure:"s3ForcePathStyle"`
+	DisableSSL          *bool          `mapstructure:"disableSSL"`
```

Comment:
We are ignoring all three options now, do we need them for other purposes?

Reply:
These have been moved to `getClient` function

Reply:
do we need them in this struct is what I was asking